src/htmx.js

```diff
@@ -2792,6 +2792,22 @@ return (function () {
                 var pushedUrl = xhr.getResponseHeader("HX-Push");
             }
 
+            if (hasHeader(xhr, /HX-Location:/i)) {
+                saveCurrentPageToHistory();
+                var redirectPath = xhr.getResponseHeader("HX-Location");
+                var swapSpec;
+                if (redirectPath.indexOf("{") === 0) {
+                    swapSpec = parseJSON(redirectPath);
```

Comment:
what if we supported a plain path too, so do this in the case of a JSON value or keep the target, etc. in the case of a plain path string?